File: app/src/main/java/com/iceteaviet/fastfoodfinder/ui/settings/StoreFilterDialogFragment.java
package com.iceteaviet.fastfoodfinder.ui.settings;

import android.animation.Animator;
import android.animation.AnimatorInflater;
import android.animation.AnimatorListenerAdapter;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.iceteaviet.fastfoodfinder.R;
import com.iceteaviet.fastfoodfinder.ui.custom.HorizontalFlowLayout;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by taq on 8/12/2016.
 */

public class StoreFilterDialogFragment extends DialogFragment {

    private static final String KEY_CIRCLE_K = "circle_k";
    private static final String KEY_MINI_STOP = "mini_stop";
    private static final String KEY_FAMILY_MART = "family_mark";
    private static final String KEY_BSMART = "bsmart";
    private static final String KEY_SHOP_N_GO = "shop_n_go";
    private static final String[] LIST_STORES = new String[]{KEY_BSMART, KEY_CIRCLE_K, KEY_FAMILY_MART, KEY_MINI_STOP, KEY_SHOP_N_GO};

    @BindView(R.id.tag_container)
    HorizontalFlowLayout tagContainer;
    @BindView(R.id.btnDone)
    Button btnDone;
    @BindView(R.id.btnCancel)
    Button btnCancel;

    public static StoreFilterDialogFragment newInstance() {
        Bundle args = new Bundle();

        StoreFilterDialogFragment fragment = new StoreFilterDialogFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.dialog_store_filter, container);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ButterKnife.bind(this, view);

        getDialog().setTitle(R.string.subscription);
        setupTagContainer();

        btnCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
            }
        });
        btnDone.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
            }
        });
    }

    @Override
    public void onActivityCreated(Bundle arg0) {
        super.onActivityCreated(arg0);
        getDialog().getWindow().getAttributes().windowAnimations = R.style.DialogAnimationUpDown;
    }

    private void setupTagContainer() {
        LayoutInflater inflater = LayoutInflater.from(getContext());
        for (String key : LIST_STORES) {
            View view = inflater.inflate(R.layout.view_store_tag, tagContainer, false);
            TagViewHolder holder = new TagViewHolder(view);
            holder.setName(getStoreName(key));
            tagContainer.addView(view);
        }
    }

    private String getStoreName(String key) {
        switch (key) {
            case KEY_CIRCLE_K:
                return "Cirle K";
            case KEY_BSMART:
                return "B’s mart";
            case KEY_FAMILY_MART:
                return "Family mart";
            case KEY_MINI_STOP:
                return "Ministop";
            case KEY_SHOP_N_GO:
                return "Shop & Go";
            default:
                return "Unknown";
        }
    }

    public class TagViewHolder {

        View itemView;
        @BindView(R.id.tag)
        TextView tvTag;

        public TagViewHolder(View view) {
            itemView = view;
            ButterKnife.bind(this, view);
            tvTag.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Animator animator = AnimatorInflater.loadAnimator(getContext(), R.animator.zoom_in_out);
                    animator.setTarget(itemView);
                    animator.setDuration(100);
                    animator.addListener(new AnimatorListenerAdapter() {
                        @Override
                        public void onAnimationStart(Animator animation) {
                            Toast.makeText(getContext(), "Successfully", Toast.LENGTH_SHORT).show();
                        }
                    });
                    animator.start();
                    v.setSelected(!v.isSelected());
                }
            });
        }

        public void setName(String name) {
            tvTag.setText(name);
        }
    }
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/ui/settings/StoreFilterDialogFragment.kt
package com.iceteaviet.fastfoodfinder.ui.settings

import android.animation.Animator
import android.animation.AnimatorInflater
import android.animation.AnimatorListenerAdapter
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.annotation.Nullable
import androidx.fragment.app.DialogFragment
import com.iceteaviet.fastfoodfinder.R
import com.iceteaviet.fastfoodfinder.ui.custom.HorizontalFlowLayout
import kotlinx.android.synthetic.main.dialog_store_filter.*
import kotlinx.android.synthetic.main.view_store_tag.view.*

/**
 * Created by taq on 8/12/2016.
 */

class StoreFilterDialogFragment : DialogFragment() {
    lateinit var tagContainer: HorizontalFlowLayout

    @Nullable
    override fun onCreateView(inflater: LayoutInflater, @Nullable container: ViewGroup?, @Nullable savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.dialog_store_filter, container)
    }

    override fun onViewCreated(view: View, @Nullable savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        tagContainer = tag_container

        dialog.setTitle(R.string.subscription)
        setupTagContainer()

        btnCancel!!.setOnClickListener { dismiss() }
        btnDone!!.setOnClickListener { dismiss() }
    }

    override fun onActivityCreated(arg0: Bundle?) {
        super.onActivityCreated(arg0)
        dialog.window!!.attributes.windowAnimations = R.style.DialogAnimationUpDown
    }

    private fun setupTagContainer() {
        val inflater = LayoutInflater.from(context)
        for (key in LIST_STORES) {
            val view = inflater.inflate(R.layout.view_store_tag, tagContainer, false)
            val holder = TagViewHolder(view)
            holder.setName(getStoreName(key))
            tagContainer.addView(view)
        }
    }

    private fun getStoreName(key: String): String {
        return when (key) {
            KEY_CIRCLE_K -> "Cirle K"
            KEY_BSMART -> "B’s mart"
            KEY_FAMILY_MART -> "Family mart"
            KEY_MINI_STOP -> "Ministop"
            KEY_SHOP_N_GO -> "Shop & Go"
            else -> "Unknown"
        }
    }

    inner class TagViewHolder internal constructor(private val itemView: View) {
        val tag: TextView = itemView.tv_tag

        init {
            tag.setOnClickListener { v ->
                val animator = AnimatorInflater.loadAnimator(context, R.animator.zoom_in_out)
                animator.setTarget(itemView)
                animator.duration = 100
                animator.addListener(object : AnimatorListenerAdapter() {
                    override fun onAnimationStart(animation: Animator) {
                        Toast.makeText(context, R.string.successfully, Toast.LENGTH_SHORT).show()
                    }
                })
                animator.start()
                v.isSelected = !v.isSelected
            }
        }

        fun setName(name: String) {
            tag.text = name
        }
    }

    companion object {
        private const val KEY_CIRCLE_K = "circle_k"
        private const val KEY_MINI_STOP = "mini_stop"
        private const val KEY_FAMILY_MART = "family_mark"
        private const val KEY_BSMART = "bsmart"
        private const val KEY_SHOP_N_GO = "shop_n_go"

        private val LIST_STORES = arrayOf(KEY_BSMART, KEY_CIRCLE_K, KEY_FAMILY_MART, KEY_MINI_STOP, KEY_SHOP_N_GO)

        fun newInstance(): StoreFilterDialogFragment {
            val args = Bundle()

            val fragment = StoreFilterDialogFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
